R/estimate_delay.R

```diff
@@ -178,24 +179,28 @@ bootstrapped_dist_fit <- function(values, dist = "lognormal",
   ## Filter out negative values
   values <- values[values >= 0]
 
-  get_single_dist <- function(values, samples = 1) {
+  get_single_dist <- function(values, samples = samples,
+                              single_sample_size = 1) {
     set_dt_single_thread()
 
     fit <- EpiNow2::dist_fit(values, samples = samples, dist = dist)
 
     out <- list()
     if (dist == "lognormal") {
-      out$meanlog <- sample(extract(fit)$mu, samples)
-      out$sdlog <- sample(extract(fit)$sigma, samples)
+      out$meanlog <- sample(extract(fit)$mu, single_sample_size)
+      out$sdlog <- sample(extract(fit)$sigma, single_sample_size)
```

Comment:
I mentioned this as a solution but I don't think its a good idea (i.e its computationally wasteful). Can we walk back to the actual bug we are trying to fix here?

Reply:
So, this change is to differentiate the two "samples" used in the scope of this function but now it differentiates how many samples are used in each fit and how many are returned per bootstrap.

I think it does the same as the original so the "computational wastefulness" is still intact? 😅 .

Reply:
> Can we walk back to the actual bug we are trying to fix here?

We want the total samples returned to be irrespective of the number of bootstraps right?

Reply:
but is that not already the case in the non-editted code?

Reply:
Somewhat. The original code tries to pass the default `sample = 1` in `get_single_dist()` to `dist_fit()` so we get the warning when it's overwritten to 1000. This new version ensures the right `sample` is passed to `dist_fit()`.

I'm unsure if the original issue is to address the warning being thrown or the number of samples being used in the summarised distribution.

Reply:
I think we want a fix for the single dist scenario vs a global refactor then?

Reply:
Two possible solutions are:
1. change the example to `samples = 10000`
2. remove the hard overwriting and warning of `N` if `N<1000`
Perhaps (2) makes more sense - are we sure 1000 is a magic number here?

Reply:
How about refactor `get_single_dist()` like so

```diff
- get_single_dist <- function(values, samples = 1) {
+ get_single_dist <- function(values, samples = samples) { 
    set_dt_single_thread()

    fit <- EpiNow2::dist_fit(values, samples = samples, dist = dist)

    out <- list()
    if (dist == "lognormal") {
-      outmeanlog<−sample(extract(fit)mu, samples)
+      outmeanlog<−sample(extract(fit)mu, 1)
-      outsdlog<−sample(extract(fit)sigma, samples)
+      outsdlog<−sample(extract(fit)sigma, 1)
  } else if (dist == "gamma") {
-      outshape<−sample(extract(fit)alpha, samples)
+     outshape<−sample(extract(fit)alpha,  1)
-      outrate<−sample(extract(fit)beta, samples)
+      outrate<−sample(extract(fit)beta, 1)
    }
    return(out)
  }
```

Reply:
I don't think I'm following what's happening here - what is `samples` (which doesn't seem to be used)?

Reply:
It's passed to `dist_fit()` so that the warnings are suppressed since the right number of samples will be used. 

From what I understand from the original code, the whole idea of `get_single_dist()` is to get only 1 sample out of the `samples` returned by `dist_fit()`.

Reply:
I think to get a total of `samples` samples we'll need to get `samples / bootstraps` from each bootstrap as @seabbs stated above. Getting one each time will only yield `bootstraps` samples.

I still think this makes most sense - why not allow fewer than 1000 samples?

> remove the hard overwriting of `N` and warning if `N<1000

Alternative would be this (what you're suggesting but with than `samples / bootstraps` samples instead of 1.

> take more samples from each bootstrap and then resample but that would be computationally wasteful

Reply:
Option 2 will still yield the warnings if the result of `samples / bootstraps` < 1000, so, I am happy to implement option 1. Not sure why it originally had a lower bound of 1000.

Reply:
Sounds good to me.